File: apps/admin-v2/src/application/use-cases/team/change-role.use-case.ts
import type { UserRepository } from '@/domain/repositories/user.repository';
import type { UserRole } from '@/domain/entities/user';

export class ChangeRoleUseCase {
  constructor(private repo: UserRepository) {}

  execute(id: string, role: UserRole) {
    return this.repo.changeRole(id, role);
  }
}

File: apps/admin_mobile/lib/application/use_cases/team/change_role_use_case.dart
import '../../../domain/entities/user.dart';
import '../../../domain/repositories/user_repository.dart';

class ChangeRoleUseCase {
  final UserRepository _repo;
  ChangeRoleUseCase(this._repo);

  Future<User> call(int id, UserRole role) => _repo.changeRole(id, role);
}
